.github/workflows/continous-integration-docker.yml

```
@@ -186,15 +232,19 @@ jobs:
         env:
           ZEBRA_TEST_LIGHTWALLETD: '1'
 
-  # Regenerate mandatory checkpoint disks.
+  # Regenerate mandatory checkpoint Zebra cached state disks.
+  #
   # Runs:
   # - on every PR update, but only if Zebra's state version changes in the PR
   # - on request
   regenerate-stateful-disks:
-    needs: [ build, changed-files]
+    needs: [ build, get-available-disks ]
     uses: ./.github/workflows/deploy-gcp-tests.yml
-    # Only run this job if the database format version has (likely) changed.
-    if: ${{ (needs.changed-files.outputs.rebuild == 'true' && github.event_name == 'push') || github.event.inputs.regenerate-disks == 'true' }}
+    # Only run this job if there's no available disk matching the actual state version from constants.rs,
+    # or if we manually execute a workflow_dispatch using regenerate-disks.
+    #
+    # Note: the output from get-available-disks should match with the caller workflow inputs
+    if: ${{ (fromJSON(needs.get-available-disks.outputs.zebra_checkpoint_disk) && github.event_name == 'push') || github.event.inputs.regenerate-disks == 'true' }}
```

Question: I'm not sure if we need the `&& github.event_name == 'push'` filter here. Why are we excluding `pull_request` events?

```suggestion
    if: ${{ fromJSON(needs.get-available-disks.outputs.zebra_checkpoint_disk) || github.event.inputs.regenerate-disks == 'true' }}
```

---

We added this as the PR was building the cached state, and then mergify was building the cached state once again.


When a commit is pushed to an opened PR (sync event) this still applies, so we're not excluding PRs as a whole. But when mergify opens a PR, it's (commonly) not adding new commits. I think we don't want to rebuild (as it takes hours) once again on the mergify queue.

---

We added this as the PR was building the cached state, and then mergify was building the cached state once again.


When a commit is pushed to an opened PR (sync event) this still applies, so we're not excluding PRs as a whole. But when mergify opens a PR, it's (commonly) not adding new commits. I think we don't want to rebuild (as it takes hours) once again on the mergify queue.

---

Nevermind, you were right.